### PR TITLE
Adds Model Extension Guide

### PR DESCRIPTION
**Pull request by kapil-varshney**
Adds Model Extension Guide

### CODE REVIEW

**Comment by kapil-varshney** [comment]
@vikashranjan26  @divyanshj16  Can you please review this guide and let me know if any information is missing or needs a better explanation?

**Comment by priyankatuteja** [comment]
@vikashranjan26  Please review this guide

File: guide/14-deep-learning/model_extension_guide.ipynb [hunk 1]
```diff
@@ -0,0 +1,567 @@
+{
```
Comment (priyankatuteja) [inline guide/14-deep-learning/model_extension_guide.ipynb:1]: <p>But, what if you came  --&gt; remove comma after but</p><p>github --&gt; GitHub</p><p>general purpose --&gt; general-purpose</p><p>Pytorch --&gt; PyTorch</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/987/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (priyankatuteja) [inline guide/14-deep-learning/model_extension_guide.ipynb:1]: <p>We need a loss function to calculate how much is the model <strong>prediction</strong> deviating from the ground truth.&nbsp;</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/987/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

**Review by priyankatuteja — APPROVED** [review]

**Comment by priyankatuteja** [comment]
@kapil-varshney Please incorporate minor changes. A very well written guide!! @vikashranjan26 Please do the code review asap

**Review by vikashranjan26 — APPROVED** [review]
LGTM

**Comment by rohitgeo** [comment]
@kapil-varshney please add subsections for segmentation and object detection for the pre- and post processing steps as the function signatures are different for those.

**Comment by mohi9282** [comment]
@kapil-varshney any updates on the requested changes? Changes need to be reviewed and merged ASAP as the guide is scheduled for V1.9.1.
cc: @AtmaMani @priyankatuteja

**Comment by mohi9282** [comment]
removing from v1.9.1 milestone per discussion with Kapil.

**Comment by kapil-varshney** [comment]
@BP-Ent  Can you help to review the guide?

**Review by BP-Ent — DISMISSED** [review]
Suggested changes made on Reviewnb

**Comment by priyankatuteja** [comment]
@kapil-varshney Please update the guide

**Comment by kapil-varshney** [comment]
@priyankatuteja Renamed the guide.

**Review by priyankatuteja — APPROVED** [review]
